File: python/base_algorithm.py
from abc import ABC, abstractmethod

class BaseAlgorithm(ABC):
    def __init__(self):
        self.name = None
        self.minimal_complexity = None
        self.average_complexity = None
        self.maximal_complexity = None

    def get_name(self):
        return self.name

    def get_minimal_complexity(self):
        return self.minimal_complexity

    def get_average_complexity(self):
        return self.average_complexity

    def get_maximal_complexity(self):
        return self.maximal_complexity

    @abstractmethod
    def get_elemental_operations(self, n):
        pass

File: src/base/BaseAlgorithm.java
package base;

import interfaces.Ialgorithm;

public abstract class BaseAlgorithm implements Ialgorithm {
    protected String name;
    protected String minimalComplexity;
    protected String averageComplexity;
    protected String maximalCompleity;

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getMinimalComplexity() {
        return minimalComplexity;
    }

    @Override
    public String getAverageComplexity() {
        return averageComplexity;
    }

    @Override
    public String getMaximalComplexity() {
        return maximalCompleity;
    }
}
